### PR TITLE
Make mount.Mount() faster^W great again

### PR DESCRIPTION
**Pull request by kolyshkin**
### pkg/mount.Mount: speedup (remove Mounted check)

This was added in PR #6669 (commit f87afda1234) because it was
otherwise impossible to do a re-mount of already mounted file system.

It is way better to just remove the Mounted() check altogether.

This change might potentially lead to multiple mounts to the same
mount point, so I audited all the users (except tests) and it looks
like no one is doing that:

 * volume/local maintains 'mounted' flag for every volume
 * pkg/chrootarchive already calls Mounted() before Mount()
   (so it actually parsed /proc/self/mountinfo twice, oops!)
 * daemon.mountVolumes() is called for docker cp only, and
   it is called once
 * daemon/graphdriver/zfs keeps track of 'mounted' status
 * daemon/graphdriver/devmapper: ditto
 * daemon.createSecretsDir() is only called once during container start

Surely I might have easily missed something so this needs a careful
review.

### pkg/archive: use mount pkg
    
It makes sense to use mount package here because
 - it no longer requires /proc to be mounted
 - it provides verbose errors so the caller doesn't have to

### TODO (not for this PR, just in general)
 * [ ] switch more users to mount.Mount/Unmount because of better errors

@thaJeztah @cpuguy83 @dmcgowan PTAL

### CODE REVIEW

**Comment by thaJeztah** [comment]
If it's possible that multiple mounts happen, should we have something similar to https://github.com/containerd/containerd/blob/master/mount/mount_linux.go#L153 ?

**Comment by kolyshkin** [comment]
> If it's possible that multiple mounts happen, should we have something similar to https://github.com/containerd/containerd/blob/master/mount/mount_linux.go#L153 ?

IDK, ideally we should not have any mounts like that at all (and I can't find any places in which we can have such mounts -- but I may overlook something).

But yes, calling umount() a few times is less evil than parsing /proc/self/mountinfo

**Review by AkihiroSuda — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
SGTM

ping @cpuguy83

File: pkg/mount/mount.go [hunk 1]
```diff
@@ -97,24 +97,19 @@ func Mounted(mountpoint string) (bool, error) {
...
-		}
-	}
 	return mount(device, target, mType, uintptr(flag), data)
 }
 
-// ForceMount will mount a filesystem according to the specified configuration,
-// *regardless* if the target path is not already mounted. Options must be
-// specified like the mount or fstab unix commands: "opt1=val1,opt2=val2". See
-// flags.go for supported option flags.
+// Mount will mount filesystem according to the specified configuration.
```
Comment (cpuguy83) [inline pkg/mount/mount.go]: `ForceMount`

Surprised the validation didn't catch this.

Comment (kolyshkin) [reply pkg/mount/mount.go]: thanks for catching this, fixed

**Comment by kolyshkin** [comment]
Fixed a nit from @cpuguy83

File: pkg/mount/mount.go (hunk 1 above)
Comment (thaJeztah) [reply pkg/mount/mount.go]: did you forget to stage? it still says `Mount` 😅☺️

Comment (kolyshkin) [reply pkg/mount/mount.go]: Grrr... please send moar :coffee:

**Comment by carlosedp** [comment]
Might need a retest... CI seems flaky :)

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by kolyshkin** [comment]
CI failure is unrelated: 

> [2020-03-11T01:25:45.060Z] fatal: unable to access 'https://github.com/docker/buildx.git/': Could not resolve host: github.com

**Comment by thaJeztah** [comment]
Yes, let's ignore ppc64le and merge

**Pull request merged**